Primo.js

```diff
@@ -212,7 +213,13 @@ function importPNX(text) {
 	
 	var itemType = ZU.xpathText(doc, '//display/type');
 	if(!itemType) {
-		throw new Error('Could not locate item type');
+		if ( ZU.xpathText(doc, '//facets/rsrctype') ) {
+			var itemType = ZU.xpathText(doc, '//facets/rsrctype');
+		} else if ( ZU.xpathText(doc, '//search/rsrctype') ) {
+			var itemType = ZU.xpathText(doc, '//search/rsrctype');
+		} else {
```

Comment:
Is this actually necessary? Every catalog I looked at had //display/type. Do you have an example? I also don't like taking metadata from facets/search for the reasons @zuphilip outlined previously.